core/src/main/java/org/wso2/testgrid/core/command/GenerateTestPlanCommand.java

```diff
@@ -487,12 +494,37 @@ private Yaml createYamlInstance() {
                 Properties configAwareInfraCombination = toConfigAwareInfrastructureCombination(
                         combination.getParameters());
                 testPlan.getInfrastructureConfig().setParameters(configAwareInfraCombination);
+
+                ScenarioConfig scenarioConfig = testgridYaml.getScenarioConfig();
+                List<ConfigChangeSet> configChangeSets = scenarioConfig.getConfigChangeSets();
+                if (configChangeSets != null) {
+                    List<TestScenario> modifiedTestScenarios = new ArrayList<>();
+                    for (ConfigChangeSet configChangeSet : configChangeSets) {
+                        for (TestScenario testScenario : scenarioConfig.getScenarios()) {
+                             TestScenario newTestScenario = new TestScenario();
+                             newTestScenario.setName(configChangeSet.getName() + ":" + testScenario.getName());
+                             newTestScenario.setDescription(testScenario.getDescription());
+                             newTestScenario.setDir(testScenario.getDir());
+                             newTestScenario.setIsPostScriptSuccessful(testScenario.isPostScriptSuccessful());
+                             newTestScenario.setIsPreScriptSuccessful(testScenario.isPreScriptSuccessful());
+                             newTestScenario.setStatus(testScenario.getStatus());
+                             newTestScenario.setTestPlan(testScenario.getTestPlan());
+                             newTestScenario.setConfigChangeSetName(configChangeSet.getName());
+                             newTestScenario.setSummaryGraphs(testScenario.getSummaryGraphs());
+                             newTestScenario.setTestCases(testScenario.getTestCases());
+                             modifiedTestScenarios.add(newTestScenario);
```

Comment:
Shouldn't you remove the old test scenarios first?

Reply:
Records are not persisted till in this line. It persist only after this step. 
Therefore, it is ok.
Problem here it already assign value for ID with the beginning of the parsing testgrid.yaml file. Therefore I had to create new ArrayList and then persist it.